awx/api/serializers.py

```diff
@@ -2970,13 +2973,14 @@ def get_field_from_model_or_attrs(fd):
 
         inventory = get_field_from_model_or_attrs('inventory')
         project = get_field_from_model_or_attrs('project')
+        fqcn_role = get_field_from_model_or_attrs('fqcn_role')
 
         if get_field_from_model_or_attrs('host_config_key') and not inventory:
             raise serializers.ValidationError({'host_config_key': _("Cannot enable provisioning callback without an inventory set.")})
 
         prompting_error_message = _("Must either set a default value or ask to prompt on launch.")
-        if project is None:
-            raise serializers.ValidationError({'project': _("Job Templates must have a project assigned.")})
+        if project is None and fqcn_role == '':
+            raise serializers.ValidationError({'project': _("Job Templates must have a project or role assigned.")})
```

Comment:
For validation - should `execution_environment` be required if `fqcn_role` is set?

Otherwise you would be using the project / default EE. While yes, that might conceptually work, in a way, it doesn't make sense. Providing the FQCN path is creating a hard dependency on the execution environment. Ultimately (even if not now), the user experience for doing this should start with the EE.

Reply:
Possibly... I think my question would be, is there a scenario where the user overrides the default EE and thus expects that default EE to be capable of role execution without the EE being supplied to the Job Template?

Reply:
Collecting details about whether or not a collection and role is available in the selected EE is probably worth persuing as an indication to the user that this job might not run with the currently selected EE. It is entirely possible that the EE could materialize the collection and role at runtime, though.